app/src/main/java/edu/stanford/spezikt/MainActivity.kt

```diff
@@ -0,0 +1,46 @@
+package edu.stanford.spezikt
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import edu.stanford.spezikt.core.designsystem.theme.SpeziKtTheme
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            SpeziKtTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+                    Greeting("SpeziKt")
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Greeting(name: String, modifier: Modifier = Modifier) {
```

Comment:
yes, we definitely need to work on that. i like the feature-based layout. but one thing i liked so far was having the preview directly in the widgets. so you could always see the changes perfectly right away. don't they do that because it violates the single responsiveness principle?